### ISSUE TITLE
中文的音频生成？

### ISSUE THREAD

**Issue by ROmanGeneratER** [yanghaha0908/EmoVoice#13]
我想测试一下EmoVoice-PP版本的checkpoint的中文语音合成效果，但是好像我生成的音素符号都是英文的，生成的结果音频也是英文的。我应该怎么正确的生成中文的音频？下面是我的相关输出信息和模型的参数配置。想请问一下是开源的checkpoint模型本身只支持英文语音合成，还是我加载的问题。
`Processing sample 1/3: EmoVoice-yanli-angry-01
/workspace/EmoVoice/src/slam_llm/utils/checkpoint_handler.py:18: DeprecationWarning: `torch.distributed._shard.checkpoint` will be deprecated, use `torch.distributed.checkpoint` instead
  from torch.distributed._shard.checkpoint import (
[2026-01-15 15:15:58][root][INFO] - train_config: {'model_name': 'tts', 'enable_ddp': False, 'enable_deepspeed': False, 'enable_fsdp': False, 'low_cpu_fsdp': False, 'run_validation': True, 'batch_size_training': 4, 'batching_strategy': 'custom', 'context_length': 4096, 'gradient_accumulation_steps': 1, 'num_epochs': 1, 'num_workers_dataloader': 2, 'warmup_steps': 1000, 'total_steps': 100000, 'validation_interval': 1000, 'lr': 0.0001, 'weight_decay': 0.0, 'gamma': 0.85, 'seed': 42, 'use_fp16': False, 'mixed_precision': True, 'val_batch_size': 1, 'use_peft': False, 'peft_config': {'peft_method': 'lora', 'r': 8, 'lora_alpha': 32, 'target_modules': ['q_proj', 'v_proj', 'k_proj', 'o_proj'], 'bias': 'none', 'task_type': 'CAUSAL_LM', 'lora_dropout': 0.05, 'inference_mode': False}, 'output_dir': 'PATH/to/save/PEFT/model', 'freeze_layers': False, 'num_freeze_layers': 1, 'quantization': False, 'one_gpu': False, 'save_model': True, 'dist_checkpoint_root_folder': 'PATH/to/save/FSDP/model', 'dist_checkpoint_folder': 'fine-tuned', 'save_optimizer': False, 'use_fast_kernels': False, 'run_test_during_validation': False, 'run_test_during_validation_file': 'test.wav', 'run_test_during_validation_prompt': '<|S2S|>', 'freeze_llm': True, 'freeze_encoder': True, 'train_embed_only': False, 'train_audio_embed_only': False, 'task_type': 'TTS', 'freeze_encoder_projector': False, 'freeze_group_decode_adapter': True, 'modeling_paradigm': 'parallel', 'interleaved_text_token_num': 12, 'interleaved_audio_token_num': 36}
[2026-01-15 15:15:58][root][INFO] - fsdp_config: {'mixed_precision': True, 'use_fp16': False, 'sharding_strategy': 'NO_SHARD', 'checkpoint_type': 'SHARDED_STATE_DICT', 'fsdp_activation_checkpointing': True, 'fsdp_cpu_offload': False, 'pure_bf16': False, 'optimizer': 'AdamW'}
[2026-01-15 15:15:58][root][INFO] - model_config: {'file': 'examples/tts/model/slam_model_tts.py:model_factory', 'llm_name': 'qwen2.5-0.5b', 'llm_path': '/workspace/models/EmoVoice/ckpt/Qwen/Qwen2.5-0.5B', 'llm_type': 'decoder_only', 'llm_dim': 896, 'encoder_name': None, 'encoder_ds_rate': 2, 'encoder_path': None, 'encoder_dim': 1280, 'encoder_projector': 'linear', 'encoder_projector_ds_rate': 5, 'modal': 'audio', 'normalize': False, 'encoder_type': 'finetune', 'vocab_config': {'text_vocabsize': 151936, 'text_specialtokens': 64, 'audio_vocabsize': 4096, 'audio_specialtokens': 64, 'code_layer': 3, 'padded_text_vocabsize': 152000, 'padded_audio_vocabsize': 4160, 'total_audio_vocabsize': 4160, 'total_vocabsize': 156160, 'eot': 151936, 'pad_t': 151937, 'input_t': 151938, 'answer_t': 151939, 'asr': 151940, 'eoa': 4096, 'pad_a': 4097, 'input_a': 4098, 'answer_a': 4099, 'split': 4100}, 'codec_decode': True, 'codec_decoder_type': 'CosyVoice', 'codec_decoder_path': '/workspace/models/EmoVoice/ckpt/CosyVoice/CosyVoice-300M-SFT', 'encoder_path_hf': None, 'code_type': 'SNAC', 'group_decode': True, 'group_decode_adapter_type': 'linear', 'whisper_decode': False, 'cosyvoice_version': 1, 'random_init': False, 'use_text_stream': False, 'save_audio_token': False, 'phn_tokenizer': '/workspace/models/EmoVoice/ckpt/Qwen/Qwen2.5-0.5B-phn'}
[2026-01-15 15:15:59][slam_llm.models.slam_model][INFO] - loading phn tokenizer from: /workspace/models/EmoVoice/ckpt/Qwen/Qwen2.5-0.5B-phn
[2026-01-15 15:16:00][slam_llm.utils.train_utils][INFO] - --> Module qwen2.5-0.5b
[2026-01-15 15:16:00][slam_llm.utils.train_utils][INFO] - --> qwen2.5-0.5b has 494.032768 Million params

[2026-01-15 15:16:00][slam_llm.utils.train_utils][INFO] - --> Module qwen2.5-0.5b
[2026-01-15 15:16:00][slam_llm.utils.train_utils][INFO] - --> qwen2.5-0.5b has 0.0 Million params

failed to import ttsfrd, use WeTextProcessing instead
/opt/conda/lib/python3.11/site-packages/diffusers/models/lora.py:393: FutureWarning: `LoRACompatibleLinear` is deprecated and will be removed in version 1.0.0. Use of `LoRACompatibleLinear` is deprecated. Please switch to PEFT backend by installing PEFT: `pip install peft`.
  deprecate("LoRACompatibleLinear", "1.0.0", deprecation_message)
[2026-01-15 15:16:08][root][INFO] - input frame rate=50
/opt/conda/lib/python3.11/site-packages/torch/nn/utils/weight_norm.py:134: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
/workspace/EmoVoice/examples/tts/utils/cosyvoice/dataset/processor.py:24: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend('soundfile')
/workspace/EmoVoice/examples/tts/utils/cosyvoice/cli/frontend.py:57: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.spk2info = torch.load(spk2info, map_location=self.device)
2026-01-15 15:16:12,649 WETEXT INFO building fst for zh_normalizer ...
[2026-01-15 15:16:12][wetext-zh_normalizer][INFO] - building fst for zh_normalizer ...
2026-01-15 15:16:48,682 WETEXT INFO done
[2026-01-15 15:16:48][wetext-zh_normalizer][INFO] - done
2026-01-15 15:16:48,682 WETEXT INFO fst path: /opt/conda/lib/python3.11/site-packages/tn/zh_tn_tagger.fst
[2026-01-15 15:16:48][wetext-zh_normalizer][INFO] - fst path: /opt/conda/lib/python3.11/site-packages/tn/zh_tn_tagger.fst
2026-01-15 15:16:48,682 WETEXT INFO           /opt/conda/lib/python3.11/site-packages/tn/zh_tn_verbalizer.fst
[2026-01-15 15:16:48][wetext-zh_normalizer][INFO] -           /opt/conda/lib/python3.11/site-packages/tn/zh_tn_verbalizer.fst
2026-01-15 15:16:48,689 WETEXT INFO found existing fst: /opt/conda/lib/python3.11/site-packages/tn/en_tn_tagger.fst
[2026-01-15 15:16:48][wetext-en_normalizer][INFO] - found existing fst: /opt/conda/lib/python3.11/site-packages/tn/en_tn_tagger.fst
2026-01-15 15:16:48,689 WETEXT INFO                     /opt/conda/lib/python3.11/site-packages/tn/en_tn_verbalizer.fst
[2026-01-15 15:16:48][wetext-en_normalizer][INFO] -                     /opt/conda/lib/python3.11/site-packages/tn/en_tn_verbalizer.fst
2026-01-15 15:16:48,689 WETEXT INFO skip building fst for en_normalizer ...
[2026-01-15 15:16:48][wetext-en_normalizer][INFO] - skip building fst for en_normalizer ...
/workspace/EmoVoice/examples/tts/utils/cosyvoice/cli/model.py:64: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.llm.load_state_dict(torch.load(llm_model, map_location=self.device), strict=True)
/workspace/EmoVoice/examples/tts/utils/cosyvoice/cli/model.py:66: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.flow.load_state_dict(torch.load(flow_model, map_location=self.device), strict=True)
/workspace/EmoVoice/examples/tts/utils/cosyvoice/cli/model.py:69: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  hift_state_dict = {k.replace('generator.', ''): v for k, v in torch.load(hift_model, map_location=self.device).items()}
[2026-01-15 15:16:51][slam_llm.utils.train_utils][INFO] - --> Module CosyVoice Codec
[2026-01-15 15:16:51][slam_llm.utils.train_utils][INFO] - --> CosyVoice Codec has 125.335343 Million params

[2026-01-15 15:16:51][slam_llm.utils.train_utils][INFO] - --> Module GroupDecodeAdapter_linear
[2026-01-15 15:16:51][slam_llm.utils.train_utils][INFO] - --> GroupDecodeAdapter_linear has 51.92928 Million params

[2026-01-15 15:16:53][slam_model_tts.py][INFO] - Resize llm embedding layer's vocab size to 156160

[2026-01-15 15:16:53][slam_model_tts.py][INFO] - loading other parts from: /workspace/models/EmoVoice/EmoVoice-PP.pt

examples/tts/model/slam_model_tts.py:58: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  ckpt_dict = torch.load(ckpt_path, map_location="cpu")
[2026-01-15 15:16:54][slam_llm.utils.train_utils][INFO] - --> Model tts
[2026-01-15 15:16:54][slam_llm.utils.train_utils][INFO] - --> tts has 0.0 Million params

[2026-01-15 15:16:54][root][INFO] - dataset_config: {'dataset': 'speech_dataset_tts', 'file': 'examples/tts/speech_dataset_tts.py:get_speech_dataset', 'train_data_path': '/tmp/temp_EmoVoice-yanli-angry-01.jsonl', 'val_data_path': '/tmp/temp_EmoVoice-yanli-angry-01.jsonl', 'train_split': 'train', 'test_split': 'validation', 'prompt': None, 'data_path': None, 'max_words': None, 'max_mel': None, 'fix_length_audio': -1, 'inference_mode': True, 'normalize': False, 'seed': 42, 'split_size': 0.1, 'vocab_config': {'text_vocabsize': 151936, 'text_specialtokens': 64, 'audio_vocabsize': 4096, 'audio_specialtokens': 64, 'code_layer': 3, 'padded_text_vocabsize': 152000, 'padded_audio_vocabsize': 4160, 'total_audio_vocabsize': 4160, 'total_vocabsize': 156160, 'eot': 151936, 'pad_t': 151937, 'input_t': 151938, 'answer_t': 151939, 'asr': 151940, 'eoa': 4096, 'pad_a': 4097, 'input_a': 4098, 'answer_a': 4099, 'split': 4100}, 'load_from_cache_file': False, 'task_type': 'TTS', 'upsampling_factor': 1, 'upsample_method': 'repeat', 'num_latency_tokens': 5, 'do_layershift': False, 'use_emo': True, 'use_text_stream': True, 'modeling_paradigm': 'parallel', 'interleaved_text_token_num': 12, 'interleaved_audio_token_num': 36}
[2026-01-15 15:16:54][root][INFO] - --> Test Set Length = 1
[2026-01-15 15:16:54][root][INFO] - decode_config: {'do_sample': False, 'max_new_tokens': 3000, 'min_length': 10, 'temperature': 1.0, 'top_k': 0, 'top_p': 1.0, 'num_beams': 1, 'num_return_sequences': 1, 'num_samples': 1, 'max_time': 0.0, 'text_repetition_penalty': 1.2, 'audio_repetition_penalty': 1.2, 'length_penalty': 1.0, 'early_stopping': False, 'no_repeat_ngram_size': 0, 'bad_words_ids': [], 'num_beam_groups': 1, 'diversity_penalty': 0.0, 'task_type': 'TTS', 'decode_text_only': False, 'streaming': False, 'stream_stride': 4, 'upsampling_factor': 1, 'input_text': False, 'do_layershift': False, 'num_latency_tokens': 5}
[2026-01-15 15:16:54][root][INFO] - Decode Strategy: Greedy
[2026-01-15 15:16:54][root][INFO] - Decode Text & Audio
[2026-01-15 15:16:54][root][INFO] - Decode Code Layer: 3
[2026-01-15 15:16:54][root][INFO] - Tone for Audio Generation: neutral_prompt_speech
[2026-01-15 15:16:54][root][INFO] - Modeling Paradigm: parallel
[2026-01-15 15:16:54][root][INFO] - ============== Start TTS Inference ==============
Generating:   0%|                                                                                              | 0/3000 [00:00<?, ?it/s]We detected that you are passing `past_key_values` as a tuple and this is deprecated and will be removed in v4.43. Please use an appropriate `Cache` class (https://huggingface.co/docs/transformers/v4.41.3/en/internal/generation_utils#transformers.Cache)
Generating: 100%|███████████████████████████████████████████████████████████████████████████████████| 3000/3000 [02:45<00:00, 18.12it/s]
[2026-01-15 15:19:40][root][INFO] - LLM Inference Time: 165.54s
[2026-01-15 15:19:40][root][INFO] - Target Text: 您是明知自己还不上还来借这个款是吗？什么暂时还不上？陈先生，您已经失业了，您还一个月消费了十六万。
[2026-01-15 15:19:40][root][INFO] - Generated Text: j uː _ ʌ n h � ə d _ ɪ n f aɪ l _ n ɛ f ᵻ t l i _ n ʌ s t iː n _ ð ʌ k s _ ð ɪ s _ w ʌ n .
[2026-01-15 15:19:40][root][WARNING] - Audio token is too long, skip. You can try to increase the max_new_tokens in the decode_config.
[2026-01-15 15:19:40][root][INFO] - ============== Inference Finished ==============
✗ Failed`

**Comment by yanghaha0908** [comment]
只开源了英文数据集训练的checkpoint
